packages/quantic/force-app/main/default/lwc/quanticRefineModalContent/quanticRefineModalContent.js

```diff
@@ -0,0 +1,204 @@
+    this.searchStatus = CoveoHeadless.buildSearchStatus(engine);
+    this.breadcrumbManager = CoveoHeadless.buildBreadcrumbManager(engine);
+    this.unsubscribeSearchStatus = this.searchStatus.subscribe(() =>
+      this.gatherFacets()
+    );
+  };
+
+  /**
+   * Gather all facets registred from the Quantic store.
+   * @returns {void}
+   */
+  gatherFacets() {
+    if (!this.facetsAreAvailable) {
+      this.data = getAllFacetsFromStore(this.engineId);
+    }
+  }
+
+  /**
+   * Deselects all the filters.
+   * @returns {void}
+   */
+  clearAllFilters() {
+    this.breadcrumbManager.deselectAll();
+  }
+
+  /**
+   * Returns facet data.
+   * @returns {Array<object>}
+   */
+  get facets() {
+    if (!this.data) {
+      return [];
+    }
+    const facetData = Object.keys(this.data).map((facetId) => {
+      if (this.data[facetId].element.localName === 'c-quantic-numeric-facet') {
+        return {
+          isNumeric: true,
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            CUMMON_FACET_PROPERTIES
+          ),
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            NUMERIC_FACET_PROPERTIES
+          ),
+        };
+      } else if (this.data[facetId].element.localName === 'c-quantic-facet') {
+        return {
+          isDefault: true,
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            CUMMON_FACET_PROPERTIES
+          ),
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            DEFAULT_FACET_PROPERTIES
+          ),
+        };
+      } else if (
+        this.data[facetId].element.localName === 'c-quantic-category-facet'
+      ) {
+        return {
+          isCategory: true,
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            CUMMON_FACET_PROPERTIES
+          ),
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            CATEGORY_FACET_PROPERTIES
+          ),
+        };
+      } else if (
+        this.data[facetId].element.localName === 'c-quantic-timeframe-facet'
+      ) {
+        return {
+          isTimeframe: true,
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            CUMMON_FACET_PROPERTIES
+          ),
+          ...this.extractFacetDataFromElement(
+            this.data[facetId].element,
+            TIMEFRAME_FACET_PROPERTIES
+          ),
+        };
+      }
+      return null;
+    });
+
+    return facetData;
+  }
+
+  /**
+   * Extracts properties of a given HTML element.
+   * @param {HTMLElement} element
+   * @param {Array<string>} properties
+   * @returns {object}
+   */
+  extractFacetDataFromElement(element, properties) {
+    const data = {};
+    properties.forEach((property) => {
+      data[property] = element[property];
+    });
+    return data;
```

Comment:
You could use `reduce` here to copy a subset of the object properties.

```suggestion
    return properties.reduce((data, property) => {
      data[property] = element[property];
      return data;
    }, {});
```